From dc9770268bb4ddd177591c80970e3bed105d61f3 Mon Sep 17 00:00:00 2001
From: Penny Zheng <Penny.Zheng@arm.com>
Date: Tue, 7 Nov 2023 16:03:07 +0800
Subject: [PATCH] tests/zephyr-mbedtls: disable failing tests

Disable failing tests, when the platform is zephyr mbedtls module.

Some AES CBC_PKCS7 related tests will fail on checking if the
output length matches the expected length, on test psa_cipher_finish.
Related issue is already tracked in
https://github.com/Mbed-TLS/mbedtls/issues/5145

Upstream-Status: Inappropriate [Kronos-specific]
Signed-off-by: Penny Zheng <penny.zheng@arm.com>
---
 api-tests/dev_apis/crypto/test_c037/test_data.h               | 4 ++++
 .../targets/tgt_dev_apis_linux/nspe/pal_crypto_config.h       | 3 +++
 2 files changed, 7 insertions(+)

diff --git a/api-tests/dev_apis/crypto/test_c037/test_data.h b/api-tests/dev_apis/crypto/test_c037/test_data.h
index d600872..38da747 100644
--- a/api-tests/dev_apis/crypto/test_c037/test_data.h
+++ b/api-tests/dev_apis/crypto/test_c037/test_data.h
@@ -78,6 +78,7 @@ static const test_data check1[] = {
 #endif
 
 #ifdef ARCH_TEST_CBC_PKCS7
+#ifndef ARCH_TEST_ON_ZEPHYR_MBEDTLS
 {
     .test_desc              = "Test psa_cipher_finish - Encrypt - AES CBC_PKCS7\n",
     .type                   = PSA_KEY_TYPE_AES,
@@ -114,6 +115,7 @@ static const test_data check1[] = {
     .expected_status        = PSA_SUCCESS
 },
 #endif
+#endif
 
 #ifdef ARCH_TEST_CIPHER_MODE_CTR
 {
@@ -295,6 +297,7 @@ static const test_data check1[] = {
     .expected_status        = PSA_SUCCESS
 },
 
+#ifndef ARCH_TEST_ON_ZEPHYR_MBEDTLS
 {
     .test_desc              = "Test psa_cipher_finish - Decrypt - AES CBC_PKCS7 (Short input)\n",
     .type                   = PSA_KEY_TYPE_AES,
@@ -313,6 +316,7 @@ static const test_data check1[] = {
     .expected_status        = PSA_SUCCESS
 },
 #endif
+#endif
 
 #ifdef ARCH_TEST_CIPHER_MODE_CTR
 {
diff --git a/api-tests/platform/targets/tgt_dev_apis_linux/nspe/pal_crypto_config.h b/api-tests/platform/targets/tgt_dev_apis_linux/nspe/pal_crypto_config.h
index 218a94c..fa505be 100755
--- a/api-tests/platform/targets/tgt_dev_apis_linux/nspe/pal_crypto_config.h
+++ b/api-tests/platform/targets/tgt_dev_apis_linux/nspe/pal_crypto_config.h
@@ -26,6 +26,9 @@
 
 #ifndef _PAL_CRYPTO_CONFIG_H_
 #define _PAL_CRYPTO_CONFIG_H_
+
+#define ARCH_TEST_ON_ZEPHYR_MBEDTLS
+
 /**
  * \def ARCH_TEST_RSA
  *
-- 
2.25.1

